### PR TITLE
provision log enhancement for rhel7

### PR DESCRIPTION
**Pull request by robin2008**
Partial address #4582 
 - clear logs in the key components during provisioning (not only debug mode, but debug mode will have more logs)
 - now only cover rhel7 stateful and stateless
 - make more syslogs into cluster.log (tag = xcat or xcat.*)
 - enhance the `msgutil_r` to make logs to file and syslog has better format
 - when install monitor die due to some plugin, the progname will be recovered.
 - the changes will not impact other distro

### CODE REVIEW

**Comment by robin2008** [comment]
UT:
case 1 - stateful:
1, `rinstall frame45cn07 osimage=<>`
2,`tail -f /var/log/xcat/cluster.log`
3, open other terminal to monitor `xcatprobe`  : `xcatprobe osdeploy -n frame45cn07`

```
Jan 17 21:31:37 fs1 xcat[96816]: xCAT: Allowing rinstall to frame45cn07 osimage for root from localhost
Jan 17 21:31:39 fs1 xcat[96817]: frame45cn07 status: powering-on statustime: 01-17-2018 21:31:39
Jan 17 21:31:59 fs1 xcat[96893]: xCAT: Allowing switchprobe -h for root from localhost
Jan 17 21:32:00 fs1 xcat[96904]: xCAT: Allowing lsdef -t site -i master -c for root from localhost
Jan 17 21:32:00 fs1 xcat[96915]: xCAT: Allowing lsdef frame45cn07 -i ip,mac -c for root from localhost
Jan 17 21:32:01 fs1 xcat[96930]: xCAT: Allowing nodels to frame45cn07 for root from localhost
Jan 17 21:36:00 fs1 xcat[96692]: frame45cn07 status: installing statustime: 01-17-2018 21:36:00
Jan 17 21:43:38 frame45cn07 xcat.deployment:  Executing post.xcat to prepare for firstbooting ...
Jan 17 21:44:13 frame45cn07 xcat.deployment:  trying to download postscripts from 192.168.3.29...
Jan 17 21:44:13 frame45cn07 xcat.deployment:  postscripts downloaded successfully
Jan 17 21:44:13 fs1 xcat[99066]: xCAT: Allowing getpostscript from frame45cn07
Jan 17 21:44:13 fs1 xcat[99069]: INFO getpostscript: Sending scripts data to frame45cn07...
Jan 17 21:44:14 frame45cn07 xcat.mypostscript:  Running postscript: syslog
Jan 17 21:44:14 frame45cn07 xcat.mypostscript:  postscript syslog return with 0
Jan 17 21:44:14 frame45cn07 xcat.mypostscript:  Running postscript: remoteshell
Jan 17 21:44:15 fs1 xcat[99079]: xCAT: Allowing getcredentials ssh_dsa_hostkey from frame45cn07
Jan 17 21:44:15 fs1 xcat[99082]: INFO credentials: sending ssh_dsa_hostkey to frame45cn07
Jan 17 21:44:15 fs1 xcat[99085]: xCAT: Allowing getcredentials ssh_rsa_hostkey from frame45cn07
Jan 17 21:44:15 fs1 xcat[99088]: INFO credentials: sending ssh_rsa_hostkey to frame45cn07
Jan 17 21:44:15 fs1 xcat[99089]: xCAT: Allowing getcredentials ssh_ecdsa_hostkey from frame45cn07
Jan 17 21:44:15 fs1 xcat[99092]: INFO credentials: sending ssh_ecdsa_hostkey to frame45cn07
Jan 17 21:44:15 fs1 xcat[99093]: xCAT: Allowing getcredentials ssh_root_pub_key:xcatdefault from frame45cn07
Jan 17 21:44:15 fs1 xcat[99096]: INFO credentials: The node (frame45cn07) is asking for sshkeys of zone: xcatdefault.
Jan 17 21:44:15 fs1 xcat[99096]: INFO credentials: sending ssh_root_pub_key to frame45cn07
Jan 17 21:44:15 fs1 xcat[99096]: INFO credentials: The root's public ssh key is in /root/.ssh/id_rsa.pub.
Jan 17 21:44:15 fs1 xcat[99097]: xCAT: Allowing getcredentials ssh_root_key:xcatdefault from frame45cn07
Jan 17 21:44:15 fs1 xcat[99100]: INFO credentials: The node (frame45cn07) is asking for sshkeys of zone: xcatdefault.
Jan 17 21:44:15 fs1 xcat[99100]: INFO credentials: sending ssh_root_key to frame45cn07
Jan 17 21:44:15 fs1 xcat[99100]: INFO credentials: The root's private ssh key is in /root/.ssh/id_rsa.
Jan 17 21:44:16 frame45cn07 xcat.mypostscript:  postscript remoteshell return with 0
Jan 17 21:44:16 frame45cn07 xcat.mypostscript:  Running postscript: syncfiles
Jan 17 21:44:16 frame45cn07 xcat.mypostscript:  postscript syncfiles return with 0
Jan 17 21:44:16 frame45cn07 xcat.mypostscript:  Running postscript: setupntp
Jan 17 21:44:25 frame45cn07 xcat.mypostscript:  postscript setupntp return with 0
Jan 17 21:44:25 frame45cn07 xcat.deployment:  finished firstboot preparation, sending request to fs1:3002 for changing status...
Jan 17 21:44:25 fs1 xcat[96692]: INFO xcatd: triggering 'nodeset frame45cn07 next'...
Jan 17 21:44:25 fs1 xcat[96692]: frame45cn07 status: booting statustime: 01-17-2018 21:44:25
Jan 17 21:44:25 fs1 xcat[96692]: xCAT: petitboot netboot: clear node(s): frame45cn07 boot device setting.
Jan 17 21:48:43 frame45cn07 xcat.xcatinstallpost:  Running /xcatpost/mypostscript.post
Jan 17 21:48:43 frame45cn07 xcat.mypostscript:  Running postbootscript: otherpkgs
Jan 17 21:48:43 frame45cn07 xcat.mypostscript:  postbootscript otherpkgs return with 0
Jan 17 21:48:43 fs1 xcat[96692]: frame45cn07 status: booted statustime: 01-17-2018 21:48:43
Jan 17 21:48:43 frame45cn07 xcat.mypostscript:  provision completed.(frame45cn07)
Jan 17 21:48:43 frame45cn07 xcat.xcatinstallpost:  /xcatpost/mypostscript.post return
```
And xcatprobe could work as well.

**Comment by robin2008** [comment]
UT case 2: stateless
1, genimage and packimage
2, `rinstall frame45cn07 osimage=<>`
3,`tail -f /var/log/xcat/cluster.log`
4, open other terminal to monitor `xcatprobe`  : `xcatprobe osdeploy -n frame45cn07`

```
Jan 17 21:59:00 fs1 xcat[114109]: xCAT: Allowing rinstall to frame45cn07 osimage=rhels7.4-ppc64le-netboot-compute for root from localhost
Jan 17 21:59:03 fs1 xcat[114110]: frame45cn07 status: powering-on statustime: 01-17-2018 21:59:03
Jan 17 22:03:19 frame45cn07 xcat.deployment:  Executing xcatroot to prepare for netbooting (dracut_33)...
Jan 17 22:03:19 frame45cn07 xcat.deployment:  Sending request to 192.168.3.29:3002 for changing status to netbooting...
Jan 17 22:03:19 fs1 xcat[96692]: frame45cn07 status: netbooting statustime: 01-17-2018 22:03:19
Jan 17 22:03:19 frame45cn07 xcat.deployment:  Downloading rootfs image from http://192.168.3.29:80//install/netboot/rhels7.4/ppc64le/compute/rootimg.cpio.gz...
Jan 17 22:03:22 frame45cn07 xcat.deployment:  Setting up RAM-root tmpfs on downloaded rootimg.cpio.[gz/xz]...
Jan 17 22:03:28 frame45cn07 xcat.deployment:  Exiting xcatroot...
Jan 17 22:03:35 frame45cn07 xcat.xcatpostinit:  Running /opt/xcat/xcatdsklspost for stateless bootstrapping...
Jan 17 22:03:36 fs1 xcat[115225]: xCAT: Allowing getpostscript from frame45cn07
Jan 17 22:03:36 fs1 xcat[115250]: INFO getpostscript: Sending scripts data to frame45cn07...
Jan 17 22:03:36 frame45cn07 xcat.mypostscript:  Running postbootscript: syslog
Jan 17 22:03:36 frame45cn07 xcat:  Install: rsyslog version 8 setup
Jan 17 22:03:36 frame45cn07 xcat.mypostscript:  postbootscript syslog return with 0
Jan 17 22:03:36 frame45cn07 xcat.mypostscript:  Running postbootscript: remoteshell
Jan 17 22:03:36 frame45cn07 xcat:  remoteshell:  setup /etc/ssh/sshd_config and ssh_config
Jan 17 22:03:36 frame45cn07 xcat:  Install: setup root .ssh
Jan 17 22:03:37 fs1 xcat[115264]: xCAT: Allowing getcredentials ssh_dsa_hostkey from frame45cn07
Jan 17 22:03:37 fs1 xcat[115267]: INFO credentials: sending ssh_dsa_hostkey to frame45cn07
Jan 17 22:03:37 frame45cn07 xcat:  remoteshell: getting ssh_host_dsa_key
Jan 17 22:03:37 fs1 xcat[115270]: xCAT: Allowing getcredentials ssh_rsa_hostkey from frame45cn07
Jan 17 22:03:37 fs1 xcat[115273]: INFO credentials: sending ssh_rsa_hostkey to frame45cn07
Jan 17 22:03:37 frame45cn07 xcat:  ssh_rsa_hostkey
Jan 17 22:03:37 fs1 xcat[115276]: xCAT: Allowing getcredentials ssh_ecdsa_hostkey from frame45cn07
Jan 17 22:03:37 fs1 xcat[115279]: INFO credentials: sending ssh_ecdsa_hostkey to frame45cn07
Jan 17 22:03:37 frame45cn07 xcat:  ssh_ecdsa_hostkey
Jan 17 22:03:37 frame45cn07 xcat:  remoteshell: gathering root ssh keys for xcatdefault
Jan 17 22:03:38 fs1 xcat[115284]: xCAT: Allowing getcredentials ssh_root_pub_key:xcatdefault from frame45cn07
Jan 17 22:03:38 fs1 xcat[115287]: INFO credentials: The node (frame45cn07) is asking for sshkeys of zone: xcatdefault.
Jan 17 22:03:38 fs1 xcat[115287]: INFO credentials: sending ssh_root_pub_key to frame45cn07
Jan 17 22:03:38 fs1 xcat[115287]: INFO credentials: The root's public ssh key is in /root/.ssh/id_rsa.pub.
Jan 17 22:03:38 frame45cn07 xcat:  remoteshell: gathering ssh_root_pub_key:xcatdefault
Jan 17 22:03:38 frame45cn07 xcat:  ssh_root_pub_key
Jan 17 22:03:38 frame45cn07 xcat:  remoteshell:sshbetweennodes is yes
Jan 17 22:03:38 fs1 xcat[115294]: xCAT: Allowing getcredentials ssh_root_key:xcatdefault from frame45cn07
Jan 17 22:03:38 fs1 xcat[115297]: INFO credentials: The node (frame45cn07) is asking for sshkeys of zone: xcatdefault.
Jan 17 22:03:38 fs1 xcat[115297]: INFO credentials: sending ssh_root_key to frame45cn07
Jan 17 22:03:38 fs1 xcat[115297]: INFO credentials: The root's private ssh key is in /root/.ssh/id_rsa.
Jan 17 22:03:38 frame45cn07 xcat:  remoteshell: gathering ssh_root_key:xcatdefault
Jan 17 22:03:38 frame45cn07 xcat:  ssh_root_key
Jan 17 22:03:38 frame45cn07 xcat:  start up sshd
Jan 17 22:03:38 frame45cn07 xcat.mypostscript:  postbootscript remoteshell return with 0
Jan 17 22:03:38 frame45cn07 xcat.mypostscript:  Running postbootscript: syncfiles
Jan 17 22:03:38 frame45cn07 xcat:  Performing syncfiles postscript
Jan 17 22:03:38 frame45cn07 xcat:  ./syncfiles: the OS name = Linux
Jan 17 22:03:38 fs1 xcat[115312]: xCAT: Allowing syncfiles from frame45cn07
Jan 17 22:03:38 frame45cn07 xcat:  ./syncfiles: Perform Syncing File action successfully
Jan 17 22:03:38 frame45cn07 xcat.mypostscript:  postbootscript syncfiles return with 0
Jan 17 22:03:38 frame45cn07 xcat.mypostscript:  Running postbootscript: setupntp
Jan 17 22:03:38 frame45cn07 xcat:  Install: Setup NTP
Jan 17 22:03:38 frame45cn07 xcat:  syncing the clock ...
Jan 17 22:03:46 frame45cn07 xcat.mypostscript:  postbootscript setupntp return with 0
Jan 17 22:03:46 frame45cn07 xcat.mypostscript:  Running postbootscript: otherpkgs
Jan 17 22:03:46 frame45cn07 xcat:  Running otherpkgs
Jan 17 22:03:46 frame45cn07 xcat.mypostscript:  postbootscript otherpkgs return with 0
Jan 17 22:03:46 frame45cn07 xcat.mypostscript:  node booted successfully, reporting status...
Jan 17 22:03:46 frame45cn07 xcat:  ready
Jan 17 22:03:46 frame45cn07 xcat:  done
Jan 17 22:03:46 fs1 xcat[96692]: frame45cn07 status: booted statustime: 01-17-2018 22:03:46
Jan 17 22:03:46 frame45cn07 xcat.mypostscript:  provision completed.(frame45cn07)
```

**Comment by robin2008** [comment]
log format in /var/log/xcat/xcat.log on compute nodes.
```
Wed Jan 17 22:03:46 EST 2018 [info]: xcat.mypostscript: postbootscript setupntp return with 0
Wed Jan 17 22:03:46 EST 2018 [info]: xcat.mypostscript: Running postbootscript: otherpkgs
  Did not install any extra packages.
Wed Jan 17 22:03:46 EST 2018 [info]: xcat.mypostscript: postbootscript otherpkgs return with 0
Wed Jan 17 22:03:46 EST 2018 [debug]: xcat.mypostscript: node booted successfully, reporting status...
Wed Jan 17 22:03:46 EST 2018 [info]: xcat.mypostscript: provision completed.(frame45cn07)
```

File: xCAT-server/share/xcat/install/scripts/post.xcat [hunk 1]
```diff
@@ -72,48 +75,46 @@ else
...
 
 if [ "$XCATDEBUGMODE" = "1" ] || [ "$XCATDEBUGMODE" = "2" ]; then
-    msgutil_r "$MASTER_IP" "debug" "/opt/xcat/xcatinfo generated" "/var/log/xcat/xcat.log"
+    msgutil_r "$MASTER_IP" "debug" "/opt/xcat/xcatinfo generated" "/var/log/xcat/xcat.log" "$log_label"
 fi
 
-
+msgutil_r "$MASTER_IP" "info" "trying to download postscripts from $MASTER_IP..." "/var/log/xcat/xcat.log" "$log_label"
 # download the postscripts
 if [ "$XCATDEBUGMODE" = "1" ] || [ "$XCATDEBUGMODE" = "2" ]; then
```
Comment (immarvin) [inline xCAT-server/share/xcat/install/scripts/post.xcat]: duplicate with line 81

File: xCAT-server/share/xcat/install/scripts/post.xcat [hunk 2]
```diff
@@ -72,48 +75,46 @@ else
...
+msgutil_r "$MASTER_IP" "info" "postscripts downloaded successfully" "/var/log/xcat/xcat.log" "$log_label"
 
 # get the precreated mypostscript file
 if [ -x /xcatpost/mypostscript ]; then
     rm -rf /xcatpost/mypostscript
 fi
 export NODE=#TABLE:nodelist:THISNODE:node#
 
+msgutil_r "$MASTER_IP" "info" "info" "trying to get mypostscript from $MASTER_IP..." "/var/log/xcat/xcat.log" "$log_label"
 if [ "$XCATDEBUGMODE" = "1" ] || [ "$XCATDEBUGMODE" = "2" ]; then
```
Comment (immarvin) [inline xCAT-server/share/xcat/install/scripts/post.xcat]: duplicate with line 109

**Comment by robin2008** [comment]
updated according to review comments.

**Comment by immarvin** [comment]
agree to merge

**Pull request merged**